docs/messaging/nats-integration.md

```diff
@@ -2,110 +2,201 @@
+[!INCLUDE [includes-hosting-and-client](../includes/includes-hosting-and-client.md)]
 
-## Prerequisites
+[NATS](https://nats.io) is a high-performance, secure, distributed messaging system. The .NET Aspire NATS integration enables you to connect to existing NATS instances, or create new instances from .NET with the [`docker.io/library/nats` container image](https://hub.docker.com/_/nats).
 
-- [Install the NATS server](https://docs.nats.io/running-a-nats-service/introduction/installation)
-- The URL to access the server.
+## Hosting integration
 
-## Get started
-
-To get started with the .NET Aspire NATS integration, install the [Aspire.NATS.Net](https://www.nuget.org/packages/Aspire.NATS.Net) NuGet package in the client-consuming project, i.e., the project for the application that uses the NATS client.
+The NATS hosting integration models a NATS server as the <xref:Aspire.Hosting.ApplicationModel.NatsServerResource> type. To access this type and APIs that allow you to add it to your app host, install the [📦 Aspire.Hosting.Nats](https://www.nuget.org/packages/Aspire.Hosting.Nats) NuGet package in the [app host](xref:aspire/app-host) project.
 
 ### [.NET CLI](#tab/dotnet-cli)
 
 ```dotnetcli
-dotnet add package Aspire.NATS.Net
+dotnet add package Aspire.Hosting.Nats
 ```
 
 ### [PackageReference](#tab/package-reference)
 
 ```xml
-<PackageReference Include="Aspire.NATS.Net"
+<PackageReference Include="Aspire.Hosting.Nats"
                   Version="[SelectVersion]" />
 ```
 
 ---
 
 For more information, see [dotnet add package](/dotnet/core/tools/dotnet-add-package) or [Manage package dependencies in .NET applications](/dotnet/core/tools/dependencies).
 
-## Example usage
+### Add NATS server resource
 
-In the _:::no-loc text="Program.cs":::_ file of your projects, call the `AddNatsClient` extension method to register an `INatsConnection` to send logs and traces to NATS and the .NET Aspire Dashboard. The method takes a connection name parameter.
+In your app host project, call <xref:Aspire.Hosting.NatsBuilderExtensions.AddNats*> on the `builder` instance to add a NATS server resource:
 
 ```csharp
-builder.AddNatsClient("nats");
+var builder = DistributedApplication.CreateBuilder(args);
+
+var nats = builder.AddNats("nats");
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(nats);
+
+// After adding all resources, run the app...
 ```
 
-You can then retrieve the `INatsConnection` instance using dependency injection. For example, to retrieve the client from a service:
+When .NET Aspire adds a container image to the app host, as shown in the preceding example with the `docker.io/library/nats` image, it creates a new NATS server instance on your local machine. A reference to your NATS server (the `nats` variable) is added to the `ExampleProject`.
```

Comment:
I'm not sure what preceding example you're talking about that references a docker image? Is there a default image that's being applied or something?

Reply:
The code right before this text. The call to `AddNats` creates a NATS server container instance on your local machine.